app/Filament/Resources/AchievementSetResource/RelationManagers/GameHashesRelationManager.php

```diff
@@ -0,0 +1,112 @@
+    }
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+
+            ]);
+    }
+
+    public function table(Table $table): Table
+    {
+        /** @var AchievementSet $achievementSet */
+        $achievementSet = $this->getOwnerRecord();
+
+        return $table
+            ->columns([
+                Tables\Columns\TextColumn::make('name')
+                    ->label('File Name'),
+
+                Tables\Columns\TextColumn::make('md5')
+                    ->label('MD5')
+                    ->fontFamily(FontFamily::Mono),
+            ])
+            ->filters([
+
+            ])
+            ->actions([
+                Tables\Actions\DetachAction::make(),
+            ])
+            ->headerActions([
+                Tables\Actions\Action::make('attachHashCompatibility')
+                    ->label('Mark Hash as Incompatible')
+                    ->color('danger')
+                    ->form([
+                        Forms\Components\Placeholder::make('warning')
+                            ->label('')
+                            ->content(fn () => new HtmlString("
+                                    🔴 PLEASE BE CAREFUL. Set types (bonus, specialty, exclusive) should automatically 
+                                    handle this for you. Only use this tool if you need to make a rare exception to the rules.
+                                    When you press Submit, the hash you've selected will no longer be eligible to receive 
+                                    this set's achievements when a player loads the hash in an emulator. This should almost
+                                    always be used VERY sparingly. If you find yourself frequently using this tool,
+                                    something is wrong. When in doubt, please ask for help!
+                                "
+                            )),
+
+                        Select::make('hash')
+                            ->label('Linked Hash')
+                            ->placeholder('Select hash')
+                            ->options(
+                                $achievementSet
+                                    ->games()
+                                    ->with('hashes')
+                                    ->get()
+                                    ->pluck('hashes')
+                                    ->flatten()
+                                    ->mapWithKeys(function (GameHash $hash) {
+                                        return [$hash->id => "{$hash->name} {$hash->md5}"];
+                                    })
+                                    ->toArray()
+                            ),
+                    ])
+                    ->action(function (array $data) use ($achievementSet) {
+                        $gameHashId = (int) $data['hash'];
+
+                        $achievementSet->gameHashes()->attach($gameHashId, ['compatible' => false]);
```

Comment:
Is there any way to report the `compatible` value in the audit message? When making a hash explicitly incompatible, the audit message just says the hash was attached, which implies it should be valid.
![image](https://github.com/user-attachments/assets/952624c7-6847-4ce6-9ca3-7eb4c1ce2a57)

Reply:
With the migration added in latest which repurposes the table, this should now hopefully be more clear:

![Screenshot 2024-10-12 at 11 01 04 AM](https://github.com/user-attachments/assets/16a5486b-d8e7-417c-b952-78d8c5999a64)